tests/spread/integration/git/task.yaml

```diff
@@ -0,0 +1,157 @@
+summary: Integration tests for git
+
+environment:
+  EXTRA_SLICES: "" # No extra slices by default.
+  # test git_bins
+  RUN/bins: test_bins
+  SLICE/bins: git_bins
+  # test git_http-clone
+  RUN/http_clone: test_http_clone
+  SLICE/http_clone: git_http-clone
+  EXTRA_SLICES/http_clone: ca-certificates_data
+  # test git_ftp-clone
+  RUN/ftp_clone: test_ftp_clone
+  SLICE/ftp_clone: git_ftp-clone
+  # test git_daemon
+  RUN/daemon_clone: test_daemon_clone
+  SLICE/daemon_clone: git_daemon
+
+execute: |
+
+  git_rootfs="/"
+  function chroot_git(){
+    # execute git within chroot. We use -C to cd since
+    # we have no $SHELL to navigate with. 
+    repo="$(realpath -e --relative-to="$git_rootfs" "$PWD")"
+    chroot "$git_rootfs" /usr/bin/git -C "$repo" "$@"
+  }
+
+  function setup(){
+    mkdir -p "$1/dev/"
+    mkdir -p "$1/etc/"
+
+    # fake essentials
+    touch "$1/dev/null"
+    head -c 500 /dev/urandom > "$1/dev/urandom"
+    cp /etc/resolv.conf $1/etc/
+    mkdir "$1/root"
+
+    # configure
+    git_rootfs="$1"
+    pushd "$1"
+
+  }
+
+  function cleanup(){
+    popd
+  }
+
+  function test_bins(){
+    chroot_git init
+    chroot_git config user.email "root@localhost"
+    chroot_git config user.name "Test Runner"
+
+    echo "Test" > "$1/test.txt"
+    chroot_git add test.txt
+    chroot_git commit -m 'test commit'
+
+    chroot_git --no-pager status
+    chroot_git --no-pager log
+
+    chroot_git blame -L 1,1 test.txt
+    [[ $(chroot_git ls-files) == "test.txt" ]]
+
+    chroot_git checkout -b test
+
+    chroot_git --no-pager diff master
+    chroot_git branch
+    chroot_git checkout master
+    chroot_git merge test
+  }
+
+  function test_http_clone(){
+    chroot_git clone --depth 1 https://git.launchpad.net/ubuntu/+source/hello hello
+    cd $_
+
+    chroot_git --no-pager log
+    chroot_git fetch origin
+    chroot_git --no-pager  branch --remotes --contains $(chroot_git rev-list -n 1 HEAD)
+  }
+
+  function test_ftp_clone(){
+    ftp_root=$(install-slices python3-pyftpdlib_libs)
```

Comment:
Do you need to install the `python3-pyftpdlib` slices in the tests or should it work by installing with apt too?

Reply:
It should work with the version installed via apt as well. I ended up chiseling this package because it was relatively simple to add for testing.